segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -478,34 +476,41 @@ public void mergeSegments(MergeSegments mergeSegments) {
 
         // Populate the AttributeUpdates for this mergeSegments operation, if any.
         AttributeUpdateCollection attributeUpdates = new AttributeUpdateCollection();
-        if (mergeSegments.getAttributeUpdates() != null) {
-            for (WireCommands.ConditionalAttributeUpdate update : mergeSegments.getAttributeUpdates()) {
-                attributeUpdates.add(new AttributeUpdate(AttributeId.fromUUID(update.getAttributeId()),
-                    AttributeUpdateType.get(update.getAttributeUpdateType()), update.getNewValue(), update.getOldValue()));
+        if (mergeSegments.getBatch().isPresent()) {
+            if (mergeSegments.getBatch().get().isLastSegment()) {
+                if (mergeSegments.getBatch().get().getSeqNo() == 1) {
+                    // There is only 1 segment in this batch
+                    // so first segment == last segment
+                    attributeUpdates.add(new AttributeUpdate(SEG_MERGE_BATCH_ID, AttributeUpdateType.ReplaceIfEquals, 0L, 0L));
```

Comment:
I would really, really like to see if we can make this method more elegant. It has a lot of if-else statements and is a bit hard to read. Could we break it down into a few methods? For example, one that generates the `SE_MERGE_BATCH_ID` and one that generates the `SEG_MERGE_SEQ_NO_IN_BATCH`? Then you'll only have to invoke 2 methods here and make the whole code more readable.

Reply:
My understanding is invoking `segmentStore.mergeStreamSegment(...)` with a non-empty attributeUpdates collection as parameter does an atomic conditional check + merge segments

Reply:
@andreipaduroiu I have fixed above code and made it more modular. Pls check...